Source/control.cpp

```diff
@@ -723,22 +726,11 @@ void DrawSpellList(const Surface &out)
 		case RSPLTYPE_SCROLL: {
 			DrawSpellCel(out, spellListItem.location, *pSpellCels, SPLICONLAST + 1);
 			strcpy(infostr, fmt::format(_("Scroll of {:s}"), _(spellDataItem.sNameText)).c_str());
-			int v = 0;
-			for (int t = 0; t < myPlayer._pNumInv; t++) {
-				if (!myPlayer.InvList[t].isEmpty()
-				    && (myPlayer.InvList[t]._iMiscId == IMISC_SCROLL || myPlayer.InvList[t]._iMiscId == IMISC_SCROLLT)
-				    && myPlayer.InvList[t]._iSpell == spellListItem.id) {
-					v++;
-				}
-			}
-			for (auto &item : myPlayer.SpdList) {
-				if (!item.isEmpty()
-				    && (item._iMiscId == IMISC_SCROLL || item._iMiscId == IMISC_SCROLLT)
-				    && item._iSpell == spellListItem.id) {
-					v++;
-				}
-			}
-			strcpy(tempstr, fmt::format(ngettext("{:d} Scroll", "{:d} Scrolls", v), v).c_str());
+			const InventoryAndBeltPlayerItemsRange items { myPlayer };
+			const int count = std::count_if(items.begin(), items.end(), [spell](const Item &item) {

```

Comment:
`count` is a bit generic for a name. Can we rename to something more semantic, like `scrollCount` maybe?

Reply:
Done